odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -0,0 +1,1064 @@
+        message = self.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'author': self.user.id,
+        })
+        self.assertEqual(message.discussion, self.discussion_2)
+        self.assertEqual(
+            message.custom_properties[0]['value'],
+            'draft',
+            msg='Should have take the default value')
+
+        # the parent come from a default value
+        self.discussion_2.message_properties = [{'id': 'test', 'type': 'char', 'default': 'default char'}]
+        default_values = self.env['test_new_api.message'].default_get(
+            ['discussion', 'custom_properties'])
+        self.assertEqual(default_values.get('discussion'), self.discussion_2.id)
+        self.assertEqual(
+            default_values.get('custom_properties', {}).get('test'),
+            'default char')
+
+        # test a default many2one
+        self.discussion_1.message_properties = [
+            {
+                'id': 'my_many2one',
+                'string': 'Partner',
+                'model': 'res.partner',
+                'type': 'many2one',
+                # send the value like the web client does
+                'default': [self.partner.id, 'Bob'],
+            },
+        ]
+        sql_definition = self._get_sql_definition(self.discussion_1)
+        self.assertEqual(sql_definition[0]['default'], self.partner.id)
+
+        read_values = self.discussion_1.read(['message_properties'])[0]['message_properties']
+        self.assertEqual(
+            read_values[0]['default'],
+            (self.partner.id, self.partner.display_name),
+            msg='When reading many2one default, it should return the display name',
+        )
+
+        # read the default many2one and deactivate the name_get
+        read_values = self.discussion_1.read(['message_properties'], load=None)[0]['message_properties']
+        self.assertEqual(
+            read_values[0]['default'],
+            self.partner.id,
+            msg='If the name_get is deactivate, it should not return the display name',
+        )
+
+        message = self.env['test_new_api.message'].create({
+            'name': 'Test Message',
+            'author': self.user.id,
+            'discussion': self.discussion_1.id,
+        })
+
+        properties = message.read(['custom_properties'])[0]['custom_properties']
+        self.assertEqual(properties[0]['value'], (self.partner.id, self.partner.display_name))
+
+        self.assertEqual(message.custom_properties[0]['value'], self.partner.id)
+
+        # give a default value and a value for a many2one
+        # the default value must be ignored
+        partner_2 = self.env['res.partner'].create({'name': 'Partner 2'})
+        property_definition = self.discussion_1.read(['message_properties'])[0]['message_properties']
+        property_definition[0]['value'] = (partner_2.id, 'Alice')
```

Comment:
Do you have to go through that logic?
Can't you simply write something like `{'partner_id': partner_id.id}` ?